src/main.rs

```diff
@@ -629,25 +667,25 @@ const APP: () = {
                         ccdr.peripheral.HRTIM,
                     );
 
-                    // IO_Update should be latched for 4 SYNC_CLK cycles after the QSPI profile
-                    // write. With pounder SYNC_CLK running at 100MHz (1/4 of the pounder reference
-                    // clock of 400MHz), this corresponds to 40ns. To accomodate rounding errors, we
-                    // use 50ns instead.
-                    //
-                    // Profile writes are always 16 bytes, with 2 cycles required per byte, coming
-                    // out to a total of 32 QSPI clock cycles. The QSPI is configured for 40MHz, so
-                    // this comes out to an offset of 800nS. We use 900ns to be safe - note that the
-                    // timer is triggered after the QSPI write, which can take approximately 120nS,
-                    // so there is additional margin.
+                    // IO_Update occurs after a fixed delay from the QSPI write. Note that the timer
+                    // is triggered after the QSPI write, which can take approximately 120nS, so
+                    // there is additional margin.
                     hrtimer.configure_single_shot(
                         hrtimer::Channel::Two,
-                        50_e-9,
-                        900_e-9,
+                        design_parameters::POUNDER_IO_UPDATE_DURATION,
+                        design_parameters::POUNDER_IO_UPDATE_DELAY,
                     );
 
                     // Ensure that we have enough time for an IO-update every sample.
+                    let sample_frequency = (design_parameters::TIMER_FREQUENCY.0
+                        as f32
+                        * 1_000_000.0)
+                        / ADC_SAMPLE_TICKS as f32;
```

Comment:
Now that `design_parameters::TIMER_FREQUENCY` is a `MegaHertz` I was wanting to rewrite it as 
```Rust
(Hertz::from(design_parameters::TIMER_FREQUENCY).0 as f32)                  
```
But it looks like the HAL implemented the `Into` traits instead of the `From` traits...I wonder if that was intentional? Convention is to implement `From` because `Into` can automatically be derived from `From`. Because of that, I think we need to use this less pretty syntax:
```Rust                                                       
(Into::<Hertz>::into(design_parameters::TIMER_FREQUENCY).0 as f32)          
```
Or perhaps
```Rust
let sample_frequency: Hertz = design_parameters::TIMER_FREQUENCY.into();
let sample_frequency = (sample_frequency.0 as f32) / (ADC_SAMPLE_TICKS as f32);
```